activesupport/lib/active_support/duration/iso8601_serializer.rb

```diff
@@ -40,7 +41,7 @@ def normalize
           end
           # If all parts are negative - let's make a negative duration
           sign = ''
-          if parts.values.all? { |v| v < 0 }
+          if !parts.empty? && parts.values.all? { |v| v < 0 }
```

Comment:
I don't think it matters whether we return a sign or not, since this is not public API. I'd prefer to avoid the unnecessary git churn.